lib/src/main/java/io/ably/lib/rest/Auth.java

```diff
@@ -409,6 +403,21 @@ public JsonObject asJSON() {
 	 * @param options
 	 */
 	public TokenDetails authorize(TokenParams params, AuthOptions options) throws AblyException {
+		return authorize(params, options, true);
+	}
+
+	/**
+	 * Alias of authorize() (0.9 RSA10l)
+	 */
+	public TokenDetails authorise(TokenParams params, AuthOptions options) throws AblyException {
+		Log.w(TAG, "authorise() alias will be removed in 1.0");
+		return authorize(params, options);
+	}
+
+	/**
+	 * Internal authorize() where the third param chooses whether to force reauthorization.
+	 */
+	public TokenDetails authorize(TokenParams params, AuthOptions options, boolean force) throws AblyException {
```

Comment:
Well I prefer @SimonWoolf's approach in https://github.com/ably/ably-js/pull/324/commits/7a9c2af1db9ec8ffc74b81bfb38f2335d8ccf292 whereby authorize is public and always auths, and then a new method such as `ensureValidAuthCredentials` is used by internals when a token should be created when needed.